### DISCUSSION TITLE
Set global stream_like_engine_allow_direct_select=1

### DISCUSSION THREAD

**Discussion by akbarali1** [ClickHouse/ClickHouse#58670] [Q&A] [answered]
I am using https://github.com/smi2/phpClickHouse. And I'm enabling Malutmoir to Clickhouse via Rabbit MQ. But when I try to select from phpClickHouse, it gives an error. But it works if I select set stream_like_engine_allow_direct_select=1 from the console.

How can I make stream_like_engine_allow_direct_select: all constant 1?

**Comment by tsolodov** [comment]
you can set it in user's profile(ClickHouse configuration), please check example here https://github.com/ClickHouse/ClickHouse/blob/8acc981e8c93380640554d80a8b1026a63cf52ec/tests/config/users.d/filelog.xml#L4

**Comment by akbarali1** [comment, marked as answer]
The following method helped me.

```php
 $this->database = new Client([
          'host'                           => 'host',
          'port'                           => '8123',
          'username'                       => 'username',
          'password'                       => 'password',
     ]);
$this->database->setTimeout(120);
$this->database->settings()->set('stream_like_engine_allow_direct_select', 1);
